pkg/gadgets/process-collector/gadget.go

```diff
@@ -73,6 +73,10 @@ func (t *Trace) Collect(trace *gadgetv1alpha1.Trace) {
 
 	if len(events) == 0 {
 		trace.Status.OperationWarning = "No container matches the requested filter"
+		// Mark the trace as complete there to avoid a timeout in CLI.
+		// Indeed, since no container matches the filter, the trace would never hit
+		// Completed state set below.
```

Comment:
IMO, it is not a workaround to avoid timeout in CLI but a bug fixing. The trace indeed completed its job, it was requested to snapshot the current sockets using a given filter but given that there is no container that matched the requested filter, it produced an empty output thus its job is done. The warning message is just additional information the trace is giving to the user to explain why the output is empty.

Reply:
I think that comment is misleading. The trace is in fact completed, it completed the tasks without producing any output because there was not any matching pod.